src/build.py

```diff
@@ -539,10 +539,22 @@ def PrintGitStatus(self):
 
 
 def RevisionModifiesFile(f):
-    # TODO: There's probably some nice single-command way to do this.
+    """Return True if the file f is modified in the index, working tree, or
+    HEAD commit."""
     if not os.path.isfile(f):
         return False
     cwd = os.path.dirname(f)
+    # If the file is modified in the index or working tree, then return true.
+    # This happens on trybots.
+    status = proc.check_output(['git', 'status', '--porcelain', f],
+                               cwd=cwd).strip()
```

Comment:
Why `--porcelain`

Reply:
That flag makes 'git status' a stable parseable output format. in git terminology, 'plumbing' commands are meant to be used by other tools (and have stable outputs), whereas 'porcelain' commands are meant to be used by users, so they have pretty but unstable outputs. I have no idea why the flag to make `git status` work like a plumbing command is `--porcelain`.